README.md

```diff
@@ -1,18 +1,120 @@
+
+The bridge was also shaped for liquidity providers who are able to assume Ethereum cost of deposit/withdrawal, as they transact amounts big enough. They will deposit on Aave Ethereum, bridge the staticATokens to Starknet and make them available for users there to buy and hold, accruing this way yield from L1. 
 
 ## Architecture
 
-![Starknet AAVE rewards architectural diagram](resources/architecture.png)
-## Testing
+![aave_bridge_5](https://user-images.githubusercontent.com/37840702/165796331-f587b34c-74a4-4954-a05c-bf55705963d6.png)
 
-The project is tested using [hardhat](https://hardhat.org/), the [starknet
-hardhat plugin](https://github.com/Shard-Labs/starknet-hardhat-plugin) and
-[starknet-devnet](https://github.com/Shard-Labs/starknet-devnet).
+## Contracts
+
+`L1`
+  * `StaticATokenLM` - an updated implementation of staticATokens which makes it possible to communicate with its respective L2 token by sending the latest `rewards_index` on token transfer or when explicitly triggered to do so.
+  *  `TokenBridge` -  handles rewards update, deposit & withdrawal of staticATokens & their underlying assets
+  *  `Proxy` - A proxy implementation 
+
+`L2`
+  * `ETHStaticAToken` - Tokens on Starknet equivalent to each staticAToken on Ethereum mainnet. Contains the same logic for tracking user rewards as staticATokens and has the same `rewards_index`.
+  * `claimable` - tracks users' claimable rewards and current reward index for each `ETHStaticAToken`
+  * `token_bridge` - is responsible for:
+    * bridging the staticATokens to and from L2. Minting and burning ETHStaticATokens on message from L1. 
+    * bridging rewAAVE token back to L1
+    * updating `rewards_index` for each ETHStaticAToken on message from L1 
+  * `rewAAVE` - an ERC20 representing the rewards on L2
+  *  `proxy` - a generic implementation of a proxy in starknet
+
+## ETHStaticATokens on L2
+
+Natively, on Aave aTokens grow in balance, not in value. To be able to create this kind of model, it is important to wrap them before bridging, converting them in a token that grows in value, not in balance.
+
+ETHStaticATokens are an implementation of the wrapped aTokens that will continuously increase in value on Starknet because they are backed by the increasing staticATokens amounts locked in the bridge contract on Ethereum. The ETHStaticATokens can then be bridged back to staticATokens.
+
+## Bridging staticATokens from L1<>L2
+
+### Transfer L1->L2: 
+
+Users can either bridge their staticAToken to L2 by calling `deposit()` on `TokenBridge`, or deposit the underlying asset of the staticAToken directly by calling `depositUnderlying()`.
+
+Calling `deposit` will result in the following:
+
+- staticATokens will be transfered from the user account to the L1 bridge
+- A `deposit` event will be emitted with the L1 token address, the recipient address on L2, and the amount
```

Comment:
let's be consistent with the naming : "with L1 token address, L2 recipient address and the amount"